### PR TITLE
add: expand Rhizomorph traversal coverage tests

### PR DESCRIPTION
**Pull request by cjprybol**
Automated polecat work.

## Summary by CodeRabbit

* **Tests**
  * Expanded test coverage for sequence generation functionality with unweighted graphs and constrained start vertices.
  * Added test cases for start-vertex selection, including edge cases with empty graphs and error handling.
  * Enhanced sequence likelihood computation tests with unweighted graph and zero-weight edge scenarios.
  * Added test suites for weighted graph conversion behavior and internal transition helper functions.

### CODE REVIEW

File: test/4_assembly/generation_test.jl [hunk 1]
```diff
@@ -101,6 +116,27 @@ Test.@testset "Generation - compute_sequence_likelihood" begin
     # A sequence with impossible k-mer transition → -Inf
     ll_impossible = Mycelia.Rhizomorph.compute_sequence_likelihood("XYZXYZ", weighted; k = 3)
     Test.@test ll_impossible == -Inf
+
+    ll_from_evidence = Mycelia.Rhizomorph.compute_sequence_likelihood("ABCDEF", graph; k = 3)
+    Test.@test ll_from_evidence == ll
```
[the remark this answers is not in the document]
Comment (cjprybol) [reply test/4_assembly/generation_test.jl]: Addressed in `8c18110` by switching this assertion to an approximate comparison with `atol = 1e-12`. I also re-ran `test/4_assembly/generation_test.jl` and `test/4_assembly/probabilistic_algorithms_next.jl` locally on the current PR head.
